### DISCUSSION TITLE
Best way to use `poetry` 1.7 without a "root project"?

### DISCUSSION THREAD

**Discussion by ilyagr** [python-poetry/poetry#8622] [Q&A] [answered]
I'm using `poetry` as a package manager to build docs for a non-Python project. Here is [the current `pyproject.toml`](https://github.com/martinvonz/jj/blob/8e4b6dfeaa62245ff973477cee7476cb9f654fed/pyproject.toml). The idea is that the user would install `poetry`, then do `poetry install` which would take care of installing the right version of MkDocs and other dependencies, and finally do `poetry run -- mkdocs serve`.

After updating to Poetry 1.7, `poetry install` results in an annoying red warning saying that it couldn't find the (non-existent) `jj-docs` project, and suggesting running `poetry install --no-root` (I think it was added in line 187 of install.py in https://github.com/python-poetry/poetry/commit/144951fa71d86ee09ddd6dd1a5782639530312f0#diff-dad751aeb5082cf19ac07d928dab9313f9fe1b1c3d0eebe2bcfb9d8efd076a55R185). It still *works*, actually, but it doesn't *look* like it worked, so I think it will confuse users.

Of course, we can tell everyone to use `poetry install --no-root` instead, but a) this is unnecessary for older versions of poetry and b) it's not elegant.

Is there any way to avoid that? Could I somehow tell `poetry` that there is no "root project", it shouldn't bother looking for it, and that a user doing `poetry install` without `--no-root` is fine?

Thanks!

**Comment by radoering** [comment, marked as answer]
I don't think so. Related issue:  #1132

**Reply by ilyagr** [reply]
I see, thank you very much for looking at this and for the link!

I'm not sure whether to mark this as "answered", since there is an answer but no actual solution. Feel free to mark this as "answered" if you prefer.
